pkg/volume/validation/pv_validation.go

```diff
@@ -53,3 +57,15 @@ func checkMountOption(pv *api.PersistentVolume) field.ErrorList {
 	}
 	return allErrs
 }
+
+// ValidatePathNoBacksteps will make sure the targetPath does not have any element which is ".."
+func ValidatePathNoBacksteps(targetPath string) error {
+	parts := strings.Split(targetPath, string(os.PathSeparator))
```

Comment:
@jhorwit2 @michmike 

I haven't tested out hostPath volume mounts on Windows yet, though it seems like a reasonable strategy for splitting up the path.  I did find a path that looks like it would slip through though. 

According to https://en.wikipedia.org/wiki/Path_(computing)#MS-DOS.2FMicrosoft_Windows_style  
```C:..\File.txt``` is a valid path, which points to a file called File.txt located in the parent directory of the current directory on drive C:.

I don't know what the current working directory would be here, but this seems like a case that would have to be checked for.

This small snippet shows it in action, as well as show the first segment which performs the backstepping, though would not be caught by the validation.

```
package main

import "fmt"
import "os"
import "path/filepath" 
import "strings"

func main() {
	path := "C:..\\test.file"
	os.Create(path)
	fmt.Println(strings.Split(filepath.ToSlash(path),"/")[0])
}

```
The output to stdout is:
```C:..```

If the binary is located in ```C:\Users\bsteciuk\go\src\test\main.exe``` and I run it from ```C:\Users\bsteciuk\go\```, then it will create the file ```C:\Users\bsteciuk\test.file```